content/posts/2023-07-24-breakdown-of-http-clients-in-elixir/index.md

```diff
@@ -0,0 +1,458 @@
+---
+title: A Breakdown of HTTP Clients in Elixir
+description: |
+  This is an overview of the HTTP clients we have available in Elixir, as long as when to use each one.
+extra:
+  cover_image: cover-image.png
+---
+
+Elixir's ecosystem has quite a few **HTTP clients** at this point. But what's
+the *best one*? In this post, I want to break down a bunch of the clients we
+have available. I'll give an overview of the clients I personally like the most,
+and I'll talk about which clients are the best choice in different use cases.
+
+<!-- more -->
+
+![Cover image of a futuristic-looking box ring with a metallic snake inside of it](cover-image.png)
+
+<span class="image-caption">This is AI generated, just to be clear</span>
+
+## All Your Clients Are Belong to Us
+
+So, let's take a whirlwind tour of some HTTP clients available in Elixir. We'll talk about these:
+
+  * [Mint](#mint)
+  * [Finch](#finch)
+  * [Req](#req)
+  * [httpc](#httpc)
+  * [Tesla](#tesla)
+
+This is not a comprehensive list of all the Elixir HTTP clients, but rather a
+list of clients that I think make sense in different situation. At the end of
+this post, you'll also find a mention of other well-known clients.
+
+### Mint
+
+Let's start with [Mint][mint]. Mint is arguably the lowest-level HTTP client
+we've got in Elixir. It's essentially a *wrapper* around a raw TCP or SSL
+socket. Its job is to make the socket **aware of the network protocol**.
```

Comment:
For Mint, I would mention early on that it is a stateless / processless abstraction in that it doesn't imply any "architecture".

Reply:
👍